ntoskrnl/io/pnpmgr/pnpres.c

```diff
@@ -548,11 +561,17 @@ IopCheckResourceDescriptor(
 {
     ULONG i, ii;
     BOOLEAN Result = FALSE;
+    PCM_FULL_RESOURCE_DESCRIPTOR FullDescriptor;
+    PCM_PARTIAL_RESOURCE_DESCRIPTOR ResDesc2;
 
+    FullDescriptor = &ResourceList->List[0];
     for (i = 0; i < ResourceList->Count; i++)
     {
-        PCM_PARTIAL_RESOURCE_LIST ResList = &ResourceList->List[i].PartialResourceList;
-        for (ii = 0; ii < ResList->Count; ii++)
+        PCM_PARTIAL_RESOURCE_LIST ResList = &FullDescriptor->PartialResourceList;
+        FullDescriptor = CmiGetNextResourceDescriptor(FullDescriptor);
+
+        ResDesc2 = &ResList->PartialDescriptors[0];
+        for (ii = 0; ii < ResList->Count; ii++, ResDesc2 = CmiGetNextPartialDescriptor(ResDesc2))
         {
             PCM_PARTIAL_RESOURCE_DESCRIPTOR ResDesc2 = &ResList->PartialDescriptors[ii];
```

Comment:
Yes, but actually no :) It should have been deleted in the previous code, but I reverted the change and went back to indecing here, because that is totally allowed, as there can be only one variable sized descriptor and it must be the last one.